### PR TITLE
Add plugin/pkg/scheduler to linted packages

### LINKED ISSUE #41868: Introduce golint into verify tool chain
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
Bugs

**Kubernetes version** (use `kubectl version`):
master

**What happened**:
There's several doc, code style improvement PR recently; it'll be better if we can improve it at the original PR. golint will help to give some advise on comments; so suggest to add it into verify tool chain.

For example, the following is the output of golint for `plugin/pkg/scheduler`:

```
$golint plugin/pkg/scheduler/
plugin/pkg/scheduler/scheduler.go:43:6: exported type PodConditionUpdater should have comment or be unexported
plugin/pkg/scheduler/scheduler.go:53:1: exported method Scheduler.StopEverything should have comment or be unexported
plugin/pkg/scheduler/scheduler.go:57:1: comment on exported type Configurator should be of the form "Configurator ..." (with optional leading article)
plugin/pkg/scheduler/scheduler.go:83:1: comment on exported type Config should be of the form "Config ..." (with optional leading article)
plugin/pkg/scheduler/scheduler.go:143:1: receiver name s should be consistent with previous receiver name sched for Scheduler
plugin/pkg/scheduler/scheduler.go:147:1: receiver name s should be consistent with previous receiver name sched for Scheduler
```

### PR DESCRIPTION
**Pull request by cblecker**
**What this PR does / why we need it**:
Adds plugin/pkg/scheduler to linted packages to improve style correctness.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #41868 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by k82cn** [comment]
Would you help to also add the following dirs?

```
plugin/pkg/scheduler
plugin/pkg/scheduler/algorithm
plugin/pkg/scheduler/algorithmprovider
plugin/pkg/scheduler/api/validation
plugin/pkg/scheduler/algorithmprovider/defaults
plugin/pkg/scheduler/algorithm/predicates
plugin/pkg/scheduler/algorithm/priorities
plugin/pkg/scheduler/schedulercache
plugin/pkg/scheduler/factory
```

**Comment by cblecker** [comment]
@k82cn yup! can work on those :)

**Comment by cblecker** [comment]
There are a large number of lint fixes required for those packages. I'm chipping away at them but this will likely take a little bit before it's ready (might as well do it all under this PR).

**Comment by timothysc** [comment]
@cblecker just poke me &| @k82cn when you're ready.

**Comment by k82cn** [comment]
sure, I'll review it firstly when it's ready :).

**Comment by timothysc** [comment]
This is going to be out-of-scope for 1.6 at this point.

**Comment by k82cn** [comment]
> There are a large number of lint fixes required for those packages. I'm chipping away at them but this will likely take a little bit before it's ready (might as well do it all under this PR).

NP, please go ahead to create PR; we'd like to combine such kind of improvement :). ping me when it's ready :).

**Comment by bsalamat** [comment]
This clean-up and addition of the scheduler is useful to prevent further lints. I think we should move forward with this PR and add other directories under scheduler in subsequent PRs.

File: plugin/pkg/scheduler/scheduler.go [hunk 1]
```diff
@@ -50,12 +52,14 @@ type Scheduler struct {
 	config *Config
 }
 
+// StopEverything when called, closes the StopEverything channel to shut down
```
Comment (jayunit100) [inline plugin/pkg/scheduler/scheduler.go]: StopEverything closes the scheduler config's StopEverything channel, to shut down the Scheduler.

File: plugin/pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -80,6 +84,7 @@ type Configurator interface {
 	CreateFromKeys(predicateKeys, priorityKeys sets.String, extenders []algorithm.SchedulerExtender) (*Config, error)
 }
 
+// Config is the definition of a scheduler configuration.
```
Comment (jayunit100) [inline plugin/pkg/scheduler/scheduler.go]: `// Config is an implementation of the Scheduler's configured input data.`

File: plugin/pkg/scheduler/scheduler.go [hunk 3]
```diff
@@ -50,12 +52,14 @@ type Scheduler struct {
...
 
+// StopEverything when called, closes the StopEverything channel to shut down
+// the Scheduler.
 func (sched *Scheduler) StopEverything() {
 	close(sched.config.StopEverything)
 }
 
-// These are the functions which need to be provided in order to build a Scheduler configuration.
-// An implementation of this can be seen in factory.go.
+// Configurator is comprised of the functions needed in order to build a
```
Comment (jayunit100) [inline plugin/pkg/scheduler/scheduler.go]: nit: `Configurator defines I/O, caching, and other functionality needed to construct a new scheduler`  okay to keep as-is also, just a suggestion.

File: plugin/pkg/scheduler/scheduler.go [hunk 4]
```diff
@@ -140,26 +145,26 @@ func NewFromConfigurator(c Configurator, modifiers ...func(c *Config)) (*Schedul
 }
 
 // Run begins watching and scheduling. It starts a goroutine and returns immediately.
-func (s *Scheduler) Run() {
-	go wait.Until(s.scheduleOne, 0, s.config.StopEverything)
+func (sched *Scheduler) Run() {
```
Comment (jayunit100) [inline plugin/pkg/scheduler/scheduler.go]: agree with this change although its not strictly necessary.

Comment (cblecker) [inline plugin/pkg/scheduler/scheduler.go]: The linter caught this, and indicated it should be standardized

**Comment by cblecker** [comment]
So I started working away at this, then rebased of off master only to find more issues :P

Wrapping in all those folders, and removing a couple low hanging fruit I fixed, there are currently 140 lint issues on master, but they're spread amongst a few different paths:
```
$ hack/verify-golint.sh 2>&1 | grep -E "^plugin/pkg/scheduler" | awk -F: '{ print $1 }' | gsed -r 's|/[^/]+$||' | sort | uniq -c
   6 plugin/pkg/scheduler/algorithm
  42 plugin/pkg/scheduler/algorithm/predicates
  27 plugin/pkg/scheduler/algorithm/priorities
  43 plugin/pkg/scheduler/factory
  22 plugin/pkg/scheduler/schedulercache
```

Instead of putting all those lint fixes into one large multi-rebase PR, I'd suggest perhaps we open a new `help-wanted` issue with these specific directories, and then the work can be split up into separate PRs.

Thoughts?

**Comment by bsalamat** [comment]
@cblecker I totally agree. Multiple separate PRs are much better than a single huge PR.

File: plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 5]
```diff
@@ -14,7 +14,7 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// This is the default algorithm provider for the scheduler.
+// Package defaults is the default algorithm provider for the scheduler.
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: I think we can remove this comments.

File: plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 6]
```diff
@@ -35,13 +35,19 @@ import (
...
+	// DefaultMaxAzureDiskVolumes defines the maximum number of PD Volumes for Azure
+	// Larger Azure VMs can actually have much more disks attached.
+	// TODO We should determine the max based on VM size
 	DefaultMaxAzureDiskVolumes = 16
-	ClusterAutoscalerProvider  = "ClusterAutoscalerProvider"
-	StatefulSetKind            = "StatefulSet"
-	KubeMaxPDVols              = "KUBE_MAX_PD_VOLS"
+	// ClusterAutoscalerProvider defines the default autoscaler provider
+	ClusterAutoscalerProvider = "ClusterAutoscalerProvider"
+	// StatefulSetKind defines the default StatefulSet kind
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithmprovider/defaults/defaults.go]: I'd like to say: `StatefulSetKind defines the name of 'StatefulSet' kind`

File: plugin/pkg/scheduler/algorithmprovider/plugins.go [hunk 7]
```diff
@@ -14,9 +14,10 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// This package is used to register algorithm provider plugins.
+// Package algorithmprovider is used to register algorithm provider plugins.
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithmprovider/plugins.go]: let's remove it.

File: plugin/pkg/scheduler/algorithmprovider/plugins.go [hunk 8]
```diff
@@ -14,9 +14,10 @@ See the License for the specific language governing permissions and
...
 package algorithmprovider
 
 import (
+	// Add defaults to algorithmprovider
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithmprovider/plugins.go]: `Import defaults of algorithmprvoider for initialization.`

File: plugin/pkg/scheduler/scheduler.go (hunk 4 above)
Comment (k82cn) [inline plugin/pkg/scheduler/scheduler.go]: which version of golint? I think we used this kind of varname in lots of funcs; what's golint's output for others?

**Comment by k82cn** [comment]
@k8s-bot ok to test

**Comment by k82cn** [comment]
let's separate by directories :)

**Comment by k82cn** [comment]
@k8s-bot non-cri e2e test this

File: plugin/pkg/scheduler/scheduler.go (hunk 4 above)
Comment (cblecker) [inline plugin/pkg/scheduler/scheduler.go]: I'm not even sure how to check the version heh. I'm just install golint with `go get -u github.com/golang/lint/golint`

**Comment by cblecker** [comment]
Updated comments per review :)

File: plugin/pkg/scheduler/algorithmprovider/plugins.go [hunk 9]
```diff
@@ -14,9 +14,9 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-// This package is used to register algorithm provider plugins.
 package algorithmprovider
 
 import (
+	// Add defaults to algorithmprovider
```
Comment (k82cn) [inline plugin/pkg/scheduler/algorithmprovider/plugins.go]: Import defaults of algorithmprvoider for initialization.

**Review by k82cn — APPROVED** [review]

**Comment by cblecker** [comment]
Looks like an unrelated flake.
@k8s-bot unit test this

**Review by timothysc — APPROVED** [review]

**Comment by cblecker** [comment]
@k8s-bot cvm gce e2e test this

**Comment by cblecker** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**